cluster-autoscaler/debuggingsnapshot/debugging_snapshot.go

```diff
@@ -58,11 +61,12 @@ type DebuggingSnapshot interface {
 // Please add all new output fields in this struct. This is to make the data
 // encoding/decoding easier as the single object going into the decoder
 type DebuggingSnapshotImpl struct {
-	NodeInfo                      []*NodeInfo `json:"NodeList"`
-	UnscheduledPodsCanBeScheduled []*v1.Pod   `json:"UnscheduledPodsCanBeScheduled"`
-	Error                         string      `json:"Error,omitempty"`
-	StartTimestamp                time.Time   `json:"StartTimestamp"`
-	EndTimestamp                  time.Time   `json:"EndTimestamp"`
+	NodeInfo                      []*NodeInfo           `json:"NodeList"`
+	UnscheduledPodsCanBeScheduled []*v1.Pod             `json:"UnscheduledPodsCanBeScheduled"`
+	Error                         string                `json:"Error,omitempty"`
+	StartTimestamp                time.Time             `json:"StartTimestamp"`
+	EndTimestamp                  time.Time             `json:"EndTimestamp"`
+	TemplateNode                  []*framework.NodeInfo `json:"TemplateNode"`
```

Comment:
Making it a list looses information about which TemplateNode is for which NodeGroup. It seems to me that if you have a hundred NodeGroups, trying to manually match template nodes to each group would be a very painful exercise (some providers may have a label that includes NodeGroup name, but we shouldn't assume that).
Is there any reason why we can't just use a map here?

also nit: TemplateNode implies a singular node which is very confusing, please pluralize the name.

Reply:
We can change it to a map with a key of nodegroup.
Done.